### PR TITLE
UCIHttpEngine

### PR DESCRIPTION
**Pull request by jeremylorino**
added implementation and `config.yaml` entries to request next move from a http endpoint

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
+.venv
 config.yml
 __pycache__
 /engines/*
 !/engines/README.md
+# lichess-bot.py
```
Comment (careless25) [inline .gitignore]: Remove this

File: config.yml.default [hunk 2]
```diff
@@ -44,6 +44,9 @@ engine:                      # engine settings
 #   log: "./engines/log.txt" # save log of lczero debug output
   silence_stderr: false      # some engines (yes you, leela) are very noisy
 
+http_engine:
+  url: "http://play.lczero.org/getMoveHardcore"
```
Comment (ddugovic) [inline config.yml.default]: There's no `#` comment?

Comment (jeremylorino) [reply config.yml.default]: `http_engine` section is only used if `protocol=http` and I am the site owner ;)

File: .gitignore (hunk 1 above)
Comment (jeremylorino) [reply .gitignore]: @careless25, removed

**Comment by careless25** [comment]
So...I don't want to merge this into master because there is no standard HTTP protocol for chess engines. We could merge this onto another branch that you can maintain

**Comment by jeremylorino** [comment]
I hate to accept that response. Regardless of whether you merge or not I will still standardize my rest api interface. Hopefully you are willing to support it as noone out there has decided to.

If we push this to another branch I may as well just keep it on my fork ☹️

**Comment by gbtami** [comment]
@jeremylorino if you want to provide another leela BOT to lichess users you can use UCI. Evangelizing a new chess engine protocol in a github PR form seems a strange idea to me.

**Pull request closed without merging**